### PR TITLE
feat(api): add endpoint to get all clubs details

### PR DESCRIPTION
**Pull request by PranshuNayak**

### CODE REVIEW

File: api/api/clubs/tests.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+from api.clubs.models import Club
+
+User = get_user_model()
+
+
+class ListClubsViewTest(APITestCase):
+    """Tests ListClubView for getting the list of clubs."""
+
+    def testListClubs_noClubExistsInDb_returnsEmptyList(self):
```
Comment (Samy-33) [inline api/api/clubs/tests.py:12]: mark these tests with `@python.markers.django_db`, it ensures that a clean db is used for each test and cleans up after the test finishes: [reference](https://pytest-django.readthedocs.io/en/latest/helpers.html#pytest.mark.django_db)

Look at tests in other files and refactor them as well.

File: api/api/clubs/serializers.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import serializers
+
+from api.clubs.models import Club
+
+
+class ClubsSerializer(serializers.ModelSerializer):
```
Comment (Samy-33) [inline api/api/clubs/serializers.py:6]: Let's keep it singular, as it's used to represent a single Club.

**Review by Samy-33 — CHANGES_REQUESTED** [review]
Overall good, few small changes.

File: api/api/clubs/tests.py (hunk 1 above)
Comment (PranshuNayak) [reply api/api/clubs/tests.py:14]: > mark these tests with `@python.markers.django_db`, it ensures that a clean db is used for each test and cleans up after the test finishes: [reference](https://pytest-django.readthedocs.io/en/latest/helpers.html#pytest.mark.django_db)
> 
> Look at tests in other files and refactor them as well.

Done!

File: api/api/auth/tests.py [hunk 3]
```diff
@@ -159,6 +160,7 @@ def test_createToken_invalidAuthorizationCodeInRequest_throwsForbiddenError(
 
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
+    @pytest.mark.django_db
```
Comment (Samy-33) [inline api/api/auth/tests.py:163]: This test doesn't use a database.

**Review by Samy-33 — APPROVED** [review]
LGTM

File: api/api/auth/tests.py (hunk 3 above)
Comment (PranshuNayak) [reply api/api/auth/tests.py:163]: here it does
`tokens_count = Token.objects.count()`
`users_count = User.objects.count()`

**Pull request merged**